kubediff

```diff
@@ -1,50 +1,98 @@
+
+        parser.add_argument('--namespace',
+                            '-n',
+                            help=('Namespace to assume for objects where it is not '
+                                  'specified (default = Kubernetes default for '
+                                  'current context)'),
+                            default='default')
+
+        parser.add_argument('--json',
+                            '-j',
+                            help='output in json format',
+                            action='store_true',
+                            dest='json')
+
+        parser.add_argument('--no-error-on-diff',
+                            '-e',
+                            help='don\'t exit with 2 if diff exists',
+                            action='store_false',
+                            dest='exit_on_diff',
+                            default=True)
+
+        parser.add_argument('path', nargs='*')
+
+        self.args = parser.parse_args()
+
+        if len(self.args.path) == 0:
+            parser.print_help()
+            sys.exit(1)
+
+
 def main():
-  parser = optparse.OptionParser("""usage: %prog [options] <dir/file>...
-
-Compare yaml files in <dir> to running state in kubernetes and print the
-differences.  This is useful to ensure you have applied all your changes to the
-appropriate environment.  This tools runs kubectl, so unless your
-~/.kube/config is configured for the correct environment, you will need to
-supply the kubeconfig for the appropriate environment.""")
-  parser.add_option("--kubeconfig", help="path to kubeconfig")
-  parser.add_option("--context", help="name of kubeconfig context to use")
-  parser.add_option("--namespace",
-                    help="the namespace to assume for objects where it's not specified (default = Kubernetes default for current context)",
-                    default="")
-  parser.add_option("-j", "--json", help="output in json format", action="store_true", dest="json")
-  parser.add_option("--no-error-on-diff", help="don't exit with 2 if diff exists",
-                    action="store_false", dest="exit_on_diff", default="true")
-  (options, args) = parser.parse_args()
-  if len(args) == 0:
-    parser.print_help()
-    sys.exit(1)
-
-  printer = QuietTextPrinter()
-  if options.json:
-    printer = JSONPrinter()
-
-  config = {
-      "kubeconfig": options.kubeconfig,
-      "namespace": options.namespace,
-      "context": options.context
-  }
-
-  failed = check_files(args, printer, config)
-  if failed and options.exit_on_diff:
-    sys.exit(2)
+
+    options = ParseArgs()
+    print(options.args)
+
+    printer = QuietTextPrinter()
+    if options.args.json:
+        printer = JSONPrinter()
+
+    config = {
+        "kubeconfig": options.args.kubeconfig,
+        "namespace": options.args.namespace,
+        "context": options.args.context
+    }
+
+    failed = check_files(options.args.path, printer, config)
```

Comment:
Maybe the variable and documentation should refer to `paths` everywhere, so make it clear that you can pass multiple paths?

```python
def check_files(paths, printer, config):
  """Check all files in 'paths' for differences to a Kubernetes cluster.

  :param printer: Where differences are reported to as they are found.
  :param dict config: Contains Kubernetes parsing and access configuration.
  :return: True if there are differences, False otherwise.
  """
  differences = 0
  for path in iter_files(paths):
....
```

Reply:
@dholbach added.

Do you like the help text?